services/cloudwatch.py

```diff
@@ -2,16 +2,54 @@
 import boto3
 
 
-# client = boto3.client("")
+client = boto3.client("cloudwatch")
+logs_client = boto3.client("logs")
 
 
 def cw_loggroup_retention_period_check():
+    compliant_resources = []
+    non_compliant_resources = []
+    log_groups = logs_client.describe_log_groups()["logGroups"]
+
+    for log_group in log_groups:
+        if "retentionInDays" in log_group and log_group["retentionInDays"] < 365:
+            non_compliant_resources.append(log_group["logGroupArn"])
+        else:
+            compliant_resources.append(log_group["logGroupArn"])
```

Comment:
기존에는 retentionInDays가 몇일 이하인지 확인하는 rule이지만, 대회 채점 방식에 초점을 두어 retentionInDays가 설정 되었는지 확인하도록 변경하도록 하겠습니다.
```suggestion
    # This rule should check if `retentionInDays` is less than n days.
    # But, instead of that, this will check if the retention setting is set to "Never expire" or not
    for log_group in log_groups:
        if "retentionInDays" in log_group:
            compliant_resources.append(log_group["logGroupArn"])
        else:
            non_compliant_resources.append(log_group["logGroupArn"])
```